### PR TITLE
dialects: Updating arith to init

### PR DESCRIPTION
**Pull request by ajrichins**
swapping over arith dialect to using __init__ as opposed to static get methods. Update usages of arith in xdsl tests and experimental interpreters.  added a test for arith select op but unsure if it is actually helping
@superlopuh @math-fehr

### CODE REVIEW

File: tests/dialects/test_arith.py [hunk 1]
```diff
@@ -116,9 +117,22 @@ def test_arith_ops(self, func: type[BinaryOperation[_BinOpArgT]]):
...
+    t = Constant.from_int_and_width(1, IntegerType(1))
+    f = Constant.from_int_and_width(0, IntegerType(1))
+    select_t_op = Select(t, t, f)
+    select_f_op = Select(f, t, f)
+    select_t_op.verify_()
+    select_f_op.verify_()
+
+    # wanting to verify it actually selected the correct operand, but not sure if in correct scope
+    assert select_t_op.result.type == t.result.type
+    assert select_f_op.result.type == f.result.type
```
Comment (ajrichins) [inline tests/dialects/test_arith.py:130]: Couldn't figure out how to connect operand to the result and I saw the self verify already does this assert

File: docs/database_example.ipynb [hunk 2]
```diff
@@ -471,8 +471,16 @@
    "name": "python3"
   },
   "language_info": {
+   "codemirror_mode": {
```
Comment (superlopuh) [inline docs/database_example.ipynb:474]: Please remove this change

**Review by superlopuh — APPROVED** [review]
beautiful

**Review by math-fehr — COMMENTED** [review]
Very nice!
Do you think you could add back the `.get` as well with a `@deprecated` decorator on it, so we don't break too much code for other people? (I'll take care of removing it after updating to the next xDSL version)

**Review by math-fehr — APPROVED** [review]
Nice! Thanks a lot!

**Pull request merged**